onnxruntime/python/tools/quantization/test_calibrate.py

```diff
@@ -1,24 +1,103 @@
+        self.assertEqual(nchw_data.shape[1], 3) # checking for 3 channels for colored image
+        self.assertEqual(nchw_data.shape[2], height) # checking if resized height is correct
+        self.assertEqual(nchw_data.shape[3], width) # checking if resized width is correct
 
-        self.assertEqual(8.999529411764707, sfacs[expected])
-        self.assertEqual(23086, zpts[expected])
+    def test_load_batch(self):
+        images_folder = 'test_images'
+        session = onnxruntime.InferenceSession('augmented_test_model.onnx')
+        (samples, channels, height, width) = session.get_inputs()[0].shape
+        batch_data = calibrate.load_batch(images_folder, height, width)
+        self.assertEqual(len(batch_data.shape), 5) # for 2D images like the ones in test_images
+        self.assertEqual(batch_data.shape[0], len(os.listdir(images_folder)))
+        self.assertEqual(batch_data.shape[2], 3)
+        self.assertEqual(batch_data.shape[3], height)
+        self.assertEqual(batch_data.shape[4], width)
+
+    def test_get_intermediate_outputs(self):
+        # Creating graph
+        A = helper.make_tensor_value_info('A', TensorProto.FLOAT, [1, 1, 5, 5])
+        B = helper.make_tensor_value_info('B', TensorProto.FLOAT, [1, 1, 5, 5])
+        C = helper.make_tensor_value_info('C', TensorProto.FLOAT, [1, 1, 5, 5])
+        clip_node = onnx.helper.make_node('Clip', ['A'], ['B'], name='Clip')
+        matmul_node = onnx.helper.make_node('MatMul', ['B', 'B'], ['C'], name='MatMul')
+        graph = helper.make_graph([clip_node, matmul_node], 'test_graph_small', [A], [C])
+        model = helper.make_model(graph)
+        model_path = 'test_model_small.onnx'
+        onnx.save(model, model_path)
+
+        # Augmenting graph
+        augmented_model = calibrate.augment_graph(model)
+        augmented_model_path = 'augmented_test_model_small.onnx'
+        onnx.save(augmented_model, augmented_model_path)
+
+        # Running inference
+        images_folder = 'test_images'
+        session = onnxruntime.InferenceSession(augmented_model_path)
+        (samples, channels, height, width) = session.get_inputs()[0].shape
+        inputs = calibrate.load_batch(images_folder, height, width)
+        dict = calibrate.get_intermediate_outputs(model_path, session, inputs)
+
+        min_results, max_results = [], []
+        for file in os.listdir(images_folder):
+            image_filepath = 'test_images/' + file
+            min_results.append(session.run(["C_ReduceMin"], {'A': calibrate.load_and_resize_image(image_filepath, height, width)}))
+            max_results.append(session.run(["C_ReduceMax"], {'A': calibrate.load_and_resize_image(image_filepath, height, width)}))
+        self.assertTrue(len(dict.keys()) == 1 and 'C' in dict.keys())
+        self.assertEqual(dict['C'][0], min(min_results)[0]) # check for correct minimum of ReduceMin values
```

Comment:
what are you exactly testing here?

Reply:
Checking that the minimum of the ReduceMin values and the maximum of the ReduceMax values are being computed correctly